library/src/test/java/com/whatsapp/stringpacks/StringPacksTest.java

```diff
@@ -55,13 +59,24 @@ public void setUp() {
     } catch (IOException e) {
       Assert.fail("Test setup failure" + e);
     }
+    try {
+      InputStream inputStream = getClass().getClassLoader().getResourceAsStream("strings_zh-rTW.pack");
+      when(assetManager.open("strings_zh-rTW.pack")).thenReturn(inputStream);
```

Comment:
Are the files in the same format as specified in [README.md](https://github.com/WhatsApp/StringPacks/blob/master/library/src/test/resources/README.md)?

Please mention the content of `zh-TW`, `ha` and `ha-NG` strings.xml as done for `zh`.
Also, what changes did you make to generate `string_ha.pack` with `ha-NG`'s translations, so that later it becomes easy for someone to add/update tests to be able to generate these files without breaking tests or re-writing them all over again

Reply:
No. These files are the pack files generated by the sample app. Strings are in `sample/app/src/main/string-packs/strings/values-ha-rNG` and `sample/app/src/main/string-packs/strings/values-zh-rTW`

Ok. I will change these to the strings that in the readme file.

Reply:
@ashish29dec I changed `string_ha.pack` and specified the format in `README.md`.

When I try to initialize the app with `zh-TW` or `ha-NG` locale `stringPacks.getString(StringPacksTestData.STRING_ID);` returns null.
The only valid string is `stringPacks.getString(1);` any other integer returns null. Not sure what I missed.

Reply:
The way we generate these test `.pack` files is the same way our actual application does it.
1. Since `StringPacksTestData.STRING_ID` is 15, you must have at least ID 15 (which is `hello_world`) translation in your `app/src/main/res/values-ha-rNG/strings.xml` and `app/src/main/res/values-zh-rTW/strings.xml`
2. Also, you must have all 16 strings in your `app/src/main/res/values/strings.xml`. The ids should be same as mentioned in [README.md](https://github.com/WhatsApp/StringPacks/blob/master/library/src/test/resources/README.md#strings_zhpack). However, the values can be anything as we are not worried about English strings.
3. Have your `config.json` specify correct values to pack `ha-NG` in `ha`
4. Run the `assemble_string_packs.py` command to generate the `.pack` files.
5. Verify if the order of your generated `StringPackIds.java` is the same as specified in [README.md](https://github.com/WhatsApp/StringPacks/blob/master/library/src/test/resources/README.md#strings_zhpack)

```
{
    R.plurals.plural_list,
    R.string.hello_africa,
    R.string.hello_antartica,
    R.string.hello_arctic,
    R.string.hello_asia,
    R.string.hello_atlantic,
    R.string.hello_australia,
    R.string.hello_europe,
    R.string.hello_forests,
    R.string.hello_indian,
    R.string.hello_mountains,
    R.string.hello_north_america,
    R.string.hello_pacific,
    R.string.hello_rivers,
    R.string.hello_south_america,
    R.string.hello_world,
}
```

This should generate your correct `strings_ha.pack` and strings_zh-rTW.pack` and calling `stringPacks.getString(StringPacksTestData.STRING_ID);` should return proper values.

Reply:
Thanks for the steps! I missed looking at the StringPackIds as a reference.

I've updated the `.pack` files and added more tests.